### DISCUSSION TITLE
docker login error

### DISCUSSION THREAD

**Discussion by fabiolamicela** [orlangure/gnomock#1026] [Q&A] [answered]
Hi,

I experience a strange behaviour with gnomock when running tests on my local system. On other systems its working fine so I guess its related to a config on my machine but I cant wrap my head around this.

When running a test which uses gnomock I get the following error message:

`can't start container: can't prepare container: can't pull image: can't pull image: Error response from daemon: Please run 'docker login'`

My ~/.docker/config.json looks like this:

`"credsStore": "osxkeychain"`

When I use the option `WithUseLocalImagesFirst` and in addition pull the desired image via docker pull the test runs fine. Pulling images works fine in the terminal since I am logged in to my docker account. But the go tests with gnomock seems to ignore my docker credentials.

Those are the client values:

```
cli = {*client.Client | 0x14000199000} 
 scheme = {string} "http"
 host = {string} "unix:///var/run/docker.sock"
 proto = {string} "unix"
 addr = {string} "/var/run/docker.sock"
 basePath = {string} ""
 client = {*http.Client | 0x1400039eed0} 
 version = {string} "1.41"
 customHTTPHeaders = {map[string]string} nil
 manualOverride = {bool} false
 negotiateVersion = {bool} true
 negotiated = {bool} false
```

Any idea what could cause the issue that docker is not logged in when running the tests but the terminal is working fine?

Thanks in advance for any tips and ideas.

**Comment by orlangure** [comment, marked as answer]
Hi, thank you for using Gnomock.

For `docker login` there is [an option you should use](https://github.com/orlangure/gnomock/blob/6bb416a0047040aa7c5ea726cd5e1263a9ad6459/options.go#L291-L300) when running tests if nothing else works. You would need to modify the test slightly to get the credentials from the filesystem or someplace else.

**Reply by fabiolamicela** [reply]
Thanks for the feedback @orlangure. I was hoping there is a different solution or explenation since its specific to my local setup and does not affect other members of the team. 

I will try to use this option based on a special flag. Thank you very much
